### PR TITLE
The Merge - CCXT PRO integration into CCXT free 🔥

### PR DESCRIPTION
**Pull request by carlosmiei**
As long expected, this PR fully merges the ccxt.pro package into ccxt seamlessly, keeping all the git history (commits and contributions) of the pro repository. 

## Implementation notes:
- Backwards compatibility (ccxt usage remains as is)
- Rest and WS implementations are kept in separate files for two reasons:
	- ease the development effort and avoid huge files
	- avoid getting huge conflicts in the currently open PRs
- WS code ended up in a different module (pro)
- ~~The JS version and the async Python/PHP automatically gain access to the WS capabilities~~
- ~~Using the different import files (ccxt.js, _init_.py, async.php, ccxt.php), I followed this approach:~~
	- ~~Regular blocking code (php and python) remains as is without WS capabilities for obvious reasons~~
	- ~~If there's a WS implementation of the exchange X, that will be the one exported, since the WS extends the REST one the user gets both functionalities (ex: `./js/ws/aax.js` )~~
	- ~~Otherwise, the REST implementation is exported (ex: `./js/rest/aax.js`)~~
- Examples updated accordingly
- clis (js/py/php) updated and working for both methods (rest and ws)
- All tests updated accordingly
- Build/transpiler adapted accordingly
- Travis updated accordingly
- Docs updated accordingly
- `ccxt.d.ts` updated accordingly
- `package.json/composer.json` updated accordingly

### Warning
- To get the PHP version working properly, you should run `composer install` inside the repository to correctly initialize the autoloader

Example of usage:

**Python**

```Python
import asyncio
import ccxt.pro as ccxt  # noqa: E402

exchange = ccxt.binance({})
async def main():
    symbol = "BTC/USDT"
    balance = await exchange.fetch_trades(symbol) # Rest API

    while True:
        trades = await exchange.watch_trades(symbol) # WS api
        print(trades)

asyncio.run(main())
```

**Javascript**

```Javascript
const ccxt = require ('ccxt').pro;
async function main () {
    const exchange = new ccxt.binance ({ })
    while (true) {
        const trades = await exchange.watchTrades ('BTC/USDT')
        console.log (trades)
    }
}
main ();
```

### CODE REVIEW

**Comment by MinsungKim-BlockCrafters** [comment]
Thank you CCXT contributors so much for this PR. I am very excited to use CCXT Pro soon.

**Comment by julienbfabre** [comment]
Hi guys, this is awesome, very much looking forward to this being in the free branch. When do you expect the merge to be completed?

**Comment by carlosmiei** [comment]
> Hi guys, this is awesome, very much looking forward to this being in the free branch. When do you expect the merge to be completed?

Hello @julienbfabre, we will try our best to release in a week or so

**Comment by carlosmiei** [comment]
> Hi guys, this is awesome, very much looking forward to this being in the free branch. When do you expect the merge to be completed?

Hello @julienbfabre, we will try our best to release in a week or so

**Review by frosty00 — COMMENTED** [review]
check the import `use React\\Promise;` in this file

**Comment by frosty00** [comment]
travis was taking too long to build so i've reduced the time of `npm run force-build` from  1m41s to 15s

File: php/async/bytetrade.php [hunk 1]
```diff
@@ -447,7 +447,7 @@ public function fetch_order_book($symbol, $limit = null, $params = array ()) {
             $request['limit'] = $limit; // default = maximum = 100
         }
         $response = yield $this->marketGetDepth (array_merge($request, $params));
-        $timestamp = $this->safe_integer($response, 'timestamp');
+        $timestamp = $this->safe_value($response, 'timestamp');
```
Comment (ttodua) [inline php/async/bytetrade.php:450]: these seem to be remnants from other PRs.

File: wiki/Exchange-Markets.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ The CCXT library currently supports the following 115 cryptocurrency exchange ma
...
+| [![coinbaseprime](https://user-images.githubusercontent.com/1294454/44539184-29f26e00-a70c-11e8-868f-e907fc236a7c.jpg)](https://exchange.coinbase.com)                                            | coinbaseprime      | [Coinbase Prime](https://exchange.coinbase.com)                                          | [![API Version *](https://img.shields.io/badge/*-lightgray)](https://docs.exchange.coinbase.com/)                                                  |                                                                                                                             | [![CCXT Pro](https://img.shields.io/badge/CCXT-Pro-black)](https://ccxt.pro) |
```
Comment (ttodua) [inline wiki/Exchange-Markets.md:51]: maybe such things are also leftovers from other PRs or local builds

File: php/async/bytetrade.php (hunk 1 above)
Comment (carlosmiei) [reply php/async/bytetrade.php:450]: @ttodua that generated file is outdated, we just don't/should commit them every time

File: wiki/Exchange-Markets.md (hunk 2 above)
Comment (carlosmiei) [reply wiki/Exchange-Markets.md:51]: same thing I guess

**Comment by simonblack** [comment]
Some of the exchange interfaces import from ccxt.rest and ccxt.ws instead of ccxt and ccxt.pro, respectively.

**Comment by carlosmiei** [comment]
> Some of the exchange interfaces import from ccxt.rest and ccxt.ws instead of ccxt and ccxt.pro, respectively.

@simonblack Don't worry some of the files (*.py and *.php) are automatically generated, so they're not updated in this PR because travis will update them upon merging.

**Pull request merged**